src/chains/ethereum/ethereum/src/transaction-pool.ts

```diff
@@ -452,6 +453,37 @@ export default class TransactionPool extends Emittery<{ drain: undefined }> {
     }
     return null;
   }
+  /**
+   * Deep clones and resets the transaction pool's `executables` such that `inProgress`
+   * transactions are moved back to `pending` and all transactions are unlocked.
+   * @returns Cloned and reset copy of transaction pool's `executables`.
+   */
+  public cloneAndResetExecutables() {
+    const executables = cloneDeep(this.executables);
+    const { inProgress, pending } = executables;
+
+    pending.forEach(txsFromOrigin => {
+      txsFromOrigin.array.forEach(tx => {
+        tx.locked = false;
+      });
+    });
+
+    inProgress.forEach(transaction => {
+      transaction.locked = false;
+      const origin = transaction.from.toString();
+      if (pending.has(origin)) {
+        const txsFromOrigin = pending.get(origin);
+        txsFromOrigin.push(transaction);
+        pending.set(origin, txsFromOrigin);
+      } else {
+        const newHeap = Heap.from(transaction, byNonce);
+        pending.set(origin, newHeap);
```

Comment:
I don't think this gets hit by any tests. Can you add some?

Reply:
Done!